src/main/java/com/hibernate/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,45 @@
+package com.hibernate.controller;
+
+import com.hibernate.model.ShoppingCart;
+import com.hibernate.model.dto.mapping.DtoResponseMapper;
+import com.hibernate.model.dto.response.ShoppingCartResponseDto;
+import com.hibernate.service.MovieSessionService;
+import com.hibernate.service.ShoppingCartService;
+import com.hibernate.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> responseMapper;
+
+    public ShoppingCartController(MovieSessionService movieSessionService,
```

Comment:
let's either use `@Autowired` annotation above the constructor everywhere - or not use it at all